contracts/utils/cryptography/WebAuthn.sol

```diff
@@ -104,69 +119,63 @@ library WebAuthn {
+                // clientDataJson.length >= typeIndex + 21
+                gt(mload(clientDataJSON), add(typeIndex, 20)),
+                eq(
+                    // get 32 bytes starting at index typexIndex in clientDataJSON, and keep the leftmost 21 bytes
+                    and(mload(add(add(clientDataJSON, 0x20), typeIndex)), shl(88, not(0))),
+                    // solhint-disable-next-line quotes
+                    '"type":"webauthn.get"'
+                )
+            )
+        }
     }
 
     /**
-     * @dev Performs strict verification of a WebAuthn Authentication Assertion.
-     *
-     * Same as {verify}, but also also verifies:
-     *
-     * [start=5]
-     * 5. {validateUserVerifiedBitSet} - confirming stronger user authentication (biometrics/PIN)
-     * 6. {validateBackupEligibilityAndState}- Backup Eligibility (`BE`) and Backup State (BS) bits
-     * relationship is valid
+     * @dev Validates that the challenge in the client data JSON matches the `expectedChallenge`.
      *
-     * This strict verification is recommended for:
-     *
-     * * High-value transactions
-     * * Privileged operations
-     * * Account recovery or critical settings changes
-     * * Applications where security takes precedence over broad authenticator compatibility
+     * Step 12 in https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion[verifying an assertion].
      */
-    function verifyStrict(
-        bytes memory challenge,
-        WebAuthnAuth memory auth,
-        bytes32 qx,
-        bytes32 qy
-    ) internal view returns (bool) {
-        return
-            validateUserVerifiedBitSet(auth.authenticatorData[32]) && // 17
-            validateBackupEligibilityAndState(auth.authenticatorData[32]) && // Consistency check
-            verify(challenge, auth, qx, qy);
+    function _validateChallenge(
+        string memory clientDataJSON,
+        uint256 challengeIndex,
+        bytes memory challenge
+    ) private pure returns (bool) {
+        // solhint-disable-next-line quotes
+        string memory expectedChallenge = string.concat('"challenge":"', Base64.encodeURL(challenge), '"');
+        string memory actualChallenge = string(
+            Bytes.slice(bytes(clientDataJSON), challengeIndex, challengeIndex + bytes(expectedChallenge).length)
+        );
+
+        return Strings.equal(actualChallenge, expectedChallenge);
     }
 
     /**
      * @dev Validates that the https://www.w3.org/TR/webauthn-2/#up[User Present (UP)] bit is set.
+     *
      * Step 16 in https://www.w3.org/TR/webauthn-2/#sctn-verifying-assertion[verifying an assertion].
      *
      * NOTE: Required by WebAuthn spec but may be skipped for platform authenticators
      * (Touch ID, Windows Hello) in controlled environments. Enforce for public-facing apps.
      */
-    function validateUserPresentBitSet(bytes1 flags) internal pure returns (bool) {
+    function _validateUserPresentBitSet(bytes1 flags) private pure returns (bool) {
```

Comment:
Not opposed to make them private, but is there any strong reason?

Reply:
My argument is to reduce the ABI of the library. I would avoid exposing functions that the users don't need so that they don't get confused by "should I use that?". If its private, it gives a clear signal that its internal helpers and that users should not call them (verify does that).

Which is all motivated by: I don't expect anyone to call that directly. 

If someone really wants, they would need to know for you extract the flags from the auth already so I'm not sure they need that helper

Reply:
Note, once we go to the vanila repo, we cannot move something from internal to private (for backward compatibility reasons). But we can move something from private to internal. So I'd have as much private as possible, and eventually make it internal if we ever got motivated requests

Reply:
> I don't expect anyone to call that directly.

Yeah I'm realizing they only make sense if you want to do something like `verifyMinimal() && validateUserPresentBit()`